### PR TITLE
[RW-4569][risk=no] Add user-facing error message when user creation fails

### PR DESCRIPTION
**Pull request by gjuggler**
Currently, all we do on a server-side error during user registration is console.error the response.

This PR adds a modal dialog, similar in spirit to what we have for failed workspace creation.

Screenshot:

Screen Shot 2020-03-15 at 10 45 14 PM

---
**PR checklist**

- [X] This PR meets the Acceptance Criteria in the JIRA story
- [X] The JIRA story has been moved to Dev Review
- [X] This PR includes appropriate unit tests
- [X] I have run and tested this change locally

### CODE REVIEW

File: ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx [hunk 1]
```diff
@@ -58,3 +69,28 @@ it('should load existing profile data', async() => {
   // Ethnicity
   expect(wrapper.find('[data-test-id="dropdown-ethnicity"]').prop('value')).toEqual(Ethnicity.HISPANIC);
 });
+
+it('should handle error when creating an account', async() => {
```
Comment (blrubenstein) [inline ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx]: Will this handle all errors? It looks like it just tests the one?

File: ui/src/app/pages/login/account-creation/account-creation-survey.tsx [hunk 2]
```diff
@@ -73,6 +76,7 @@ export class AccountCreationSurvey extends React.Component<AccountCreationSurvey
   createAccount(): void {
     const {invitationKey, termsOfServiceVersion, onComplete} = this.props;
     this.setState({creatingAccount: true});
+    console.log('creating account...', invitationKey);
```
Comment (blrubenstein) [inline ui/src/app/pages/login/account-creation/account-creation-survey.tsx]: nit: remove console log

File: ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx (hunk 1 above)
Comment (gjuggler) [reply ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx]: Yes, the code handles any type of error. This test demonstrates the end-to-end flow and validates that the error message is templated into the modal dialog. I don't think explicit tests for different error codes would be worth the redundancy.

File: ui/src/app/pages/login/account-creation/account-creation-survey.tsx (hunk 2 above)
Comment (gjuggler) [reply ui/src/app/pages/login/account-creation/account-creation-survey.tsx]: Done

File: ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx (hunk 1 above)
Comment (blrubenstein) [reply ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx]: Sorry, yes, I understood the feature, I was just nitpicking the text on the test. This seems fine then.

**Pull request merged**